lib/matplotlib/backends/web_backend/mpl.js

```diff
@@ -459,20 +459,30 @@ mpl.figure.prototype.mouse_event = function(event, name) {
 }
 
 mpl.figure.prototype.key_event = function(event, name) {
-    /* Don't fire events just when a modifier is changed.  Modifiers are
-       sent along with other keys. */
-    if (event.keyCode >= 16 && event.keyCode <= 20) {
-        return;
-    }
 
     var value = '';
-    if (event.ctrlKey) {
+    if (event.ctrlKey && !(event.keyCode == 17)) {
         value += "ctrl+";
     }
-    if (event.altKey) {
+    if (event.altKey && !(event.keyCode == 18)) {
         value += "alt+";
     }
-    value += String.fromCharCode(event.keyCode).toLowerCase();
+    if (event.shiftKey && !(event.keyCode == 16)) {
+        value += "shift+";
+    }
+
+    if (event.keyCode == 17) {
+        value += "control";
+    }
+    else if (event.keyCode == 18) {
+        value += "alt";
+    }
+    else if (event.keyCode == 16) {
+        value += "shift";
+    }
+    else {
+        value += String.fromCharCode(event.keyCode).toLowerCase();
```

Comment:
I think only if the character can be capitalized.
On Jan 23, 2015 9:48 AM, "Steven Silvester" <notifications@github.com>
wrote:

> In lib/matplotlib/backends/web_backend/mpl.js
> <https://github.com/matplotlib/matplotlib/pull/4029#discussion_r23453341>:
>
> > -    value += String.fromCharCode(event.keyCode).toLowerCase();
> > +    if (event.shiftKey && !(event.keyCode == 16)) {
> > +        value += "shift+";
> > +    }
> > +
> > +    if (event.keyCode == 17) {
> > +        value += "control";
> > +    }
> > +    else if (event.keyCode == 18) {
> > +        value += "alt";
> > +    }
> > +    else if (event.keyCode == 16) {
> > +        value += "shift";
> > +    }
> > +    else {
> > +        value += String.fromCharCode(event.keyCode).toLowerCase();
>
> Well, we'd also need to remove "shift" from the value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4029/files#r23453341>.
>